[Term Entry] Create staticmethod.md | Closes issue #2387 (#2665)
Codecademy/docs: content/python/concepts/built-in-functions/terms/staticmethod/staticmethod.md

---
Title: 'staticmethod()'

> Description: 'The staticmethod() built-in function transforms a method to a static method'

Reviewer: ```suggestion
Description: 'Transforms a method to a static method.'
```

Subjects:
  - 'Computer Science'
  - 'Data Science'
Tags:
  - 'Functions'
  - 'Methods'
CatalogContent:
  - 'learn-python-3'
  - 'paths/computer-science'

>   - 'paths/data-science'

Reviewer: ```suggestion
```

Reviewer: we only list two items under catalog content

---

> The built-in function staticmethod() converts a method to a static method. The function to be converted is passed as a parameter to the function and the static conversion of the method is returned. This has several uses, one case is when you want to create a new static method from an existing instance method.

Reviewer: ```suggestion
The built-in function **`staticmethod()`** converts a method to a static method. The function to be converted is passed as a parameter and the static conversion of the method is returned. 

This function has several applications, one use case is the generation of a new static method from an existing instance method. A static method differs from a class or instance method in that it can't modify the class or instance. These methods serve to impose a tighter scope over a given functionality and are used to signal the design intention and use of the class.
```

> Note: While use of this built-in function was common in older versions of python, it is considered more Pythonic to use the @staticmethod decorator to define static methods  

Reviewer: ```suggestion
> **Note:** While use of this built-in function was common in older versions of Python, it is considered more Pythonic to use the @staticmethod decorator to define static methods.  
```

## Syntax

> ```py

Reviewer: ```suggestion
```pseudo
```

staticmethod(function)
```

## Example 1

Use `staticmethod()` to convert the print_welcome method in the `Codecademy` class to a static method:

```py
class Codecademy:
  def print_welcome(name):
    print("Welcome to...", name)

> #converting to static method

Reviewer: ```suggestion
#] Converting to static method
```

Codecademy.print_welcome = staticmethod(Codecademy.print_welcome)

> #calling as a static method by the class

Reviewer: ```suggestion
# Calling as a static method by the class
```

Codecademy.print_welcome("Codecademy")

> #calling as a static method by the object

Reviewer: ```suggestion
# Calling as a static method by the object
```

Reviewer: these should also include an output in a ```shell block

Codecademy().print_welcome("Codecademy")
```

## Example 2

Use `staticmethod()` to create a static method in `Codecademy` class from an external function:

```py
def print_hello():
    print("Hello, I'm not from the class")

class Codecademy:
  #converting print_hello to a static function and assigning to a method in Codecademy class
  codecademy_hello = staticmethod(print_hello)

#calling codecademy_hello as a static method
Codecademy.codecademy_hello()
```

## Example 3

> Use `staticmethod()` to create a static function from a method in the Codecademy class and call the function on its own:

Reviewer: ```suggestion
Use `staticmethod()` to create a static function from a method in the `Codecademy` class and call the function on its own:
```

```py
class Codecademy:
  def print_welcome(name):
    print("Welcome to...", name)

> #converting to static method

Reviewer: ```suggestion
# Converting to static method
```

new_welcome = staticmethod(Codecademy.print_welcome)

> #calling new_welcome

Reviewer: ```suggestion
# Calling new_welcome
```
